hystrix/src/test/java/feign/hystrix/HystrixBuilderTest.java

```diff
@@ -148,7 +148,7 @@ public void fallbacksApplyOnError() {
   @Test
   public void errorInFallbackHasExpectedBehavior() {
     thrown.expect(HystrixRuntimeException.class);
-    thrown.expectMessage("contributors failed and fallback failed.");
+    thrown.expectMessage("feign#public abstract java.util.List feign.hystrix.HystrixBuilderTest$GitHub.contributors(java.lang.String,java.lang.String) failed and fallback failed.");
```

Comment:
I think in related issues I've mentioned Feign.configKey is the
appropriate tool for the job.

The cheapest way is to replace your method.toString with:

Feign.configKey(method.getDeclaringClass(), method);

Later, I can help extract the Setter thing as I mentioned earlier, but
this would get us by. In fact, I'm not even sure we need the feign
prefix as it is highly unlikely anything else uses this formatting
(javadoc link format).

Feign.configKey keeps formatting consistent with other feign things
(like logs) and also has uniqueness including the simple name and the
method's parameters. If you don't think it looks better in the UI,
call me out!


On Wed, Aug 17, 2016 at 8:46 AM, Ryan Baxter <notifications@github.com> wrote:
> In hystrix/src/test/java/feign/hystrix/HystrixBuilderTest.java:
>
>> @@ -148,7 +148,7 @@ public void fallbacksApplyOnError() {
>>    @Test
>>    public void errorInFallbackHasExpectedBehavior() {
>>      thrown.expect(HystrixRuntimeException.class);
>> -    thrown.expectMessage("contributors failed and fallback failed.");
>> +    thrown.expectMessage("feign#public abstract java.util.List
>> feign.hystrix.HystrixBuilderTest$GitHub.contributors(java.lang.String,java.lang.String)
>> failed and fallback failed.");
>
> Guess I am not sure what you mean by without any gain in uniqueness? Right
> now the key is the method name now the key is the entire signature which
> makes it much more unique.
>
> What is the convention you are referring to?
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub, or mute the thread.